lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -503,44 +510,45 @@ def generate_test_result_tree(tmp_path: Path, dir_name: str) -> Path:
     class MockTarball:
         def __init__(self, path: Path, controller: Controller):
             self.name = Dataset.stem(path)
+            self.resource_id = "ABC"
             self.tarball_path = path
-            self.cache = controller.cache / "ABC"
+            self.cache = controller.cache / self.resource_id
+            self.lock = self.cache / "lock"
+            self.last_ref = self.cache / "last_ref"
             self.unpacked = None
+            self.controller = controller
 
-    def test_unpack_tar_subprocess_exception(self, make_logger, monkeypatch):
+    def test_unpack_tar_subprocess_exception(
+        self, make_logger, db_session, monkeypatch
+    ):
         """Show that, when unpacking of the Tarball fails and raises
         an Exception it is handled successfully."""
         tar = Path("/mock/A.tar.xz")
         cache = Path("/mock/.cache")
-        rmtree_called = False
-
-        def mock_rmtree(path: Path, ignore_errors=False):
-            nonlocal rmtree_called
-            rmtree_called = True
-
-            assert ignore_errors
-            assert path == cache / "ABC"
 
+        @staticmethod
         def mock_run(command, _dir_path, exception, dir_p):
             verb = "tar"
             assert command.startswith(verb)
             raise exception(dir_p, subprocess.TimeoutExpired(verb, 43))
 
         with monkeypatch.context() as m:
-            m.setattr(Path, "mkdir", lambda path, parents: None)
-            m.setattr(Tarball, "subprocess_run", staticmethod(mock_run))
-            m.setattr(shutil, "rmtree", mock_rmtree)
+            m.setattr(Path, "exists", lambda self: True)
+            m.setattr(Path, "mkdir", lambda path, parents=False, exist_ok=False: None)
+            m.setattr(Path, "touch", lambda path, exist_ok=False: None)
+            m.setattr(Tarball, "subprocess_run", mock_run)
             m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
             m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
             tb = Tarball(tar, Controller(Path("/mock/archive"), cache, make_logger))
-            with pytest.raises(TarballUnpackError) as exc:
-                tb.unpack()
+            with pytest.raises(Exception) as exc:
+                tb.cache_create()
             msg = f"An error occurred while unpacking {tar}: Command 'tar' timed out after 43 seconds"
             assert str(exc.value) == msg
             assert exc.type == TarballUnpackError
```

Comment:
I'll grant that, in the old code, the `exc.type` assertion at line 540 was redundant with the `raises` filter at line 536...but I would have left the filter in place and removed the pointless assertion...why did you relax the filter?  Wouldn't it be better to let PyTest fail the test immediately if we get the wrong exception, rather than delaying that to our custom assertions?  (The `exc.type` assertion now looks nearly pointless, as it's unlikely that the message will match in the previous assertion if the type is wrong....)